lib/earmark_parser/line_scanner.ex

```diff
@@ -265,8 +265,8 @@ defmodule EarmarkParser.LineScanner do
 
   defp _with_lookahead([line_lnb | lines], options, recursive) do
     case type_of(line_lnb, options, recursive) do
-      %Line.Fence{delimiter: delimiter, indent: 0} = fence when recursive ->
-        stop = ~r/\A (?: #{delimiter} ) \s* ([^`\s]*) \s* \z/xu
+      %Line.Fence{delimiter: delimiter, indent: 0} = fence ->
+        stop = ~r/\A (\s*) (?: #{delimiter} ) \s* ([^`\s]*) \s* \z/xu
```

Comment:
The optional whitespace is necessary when the code block is inside a list.

(Alternatively we could do this only `when not recursive`)